backend/tournesol/tests/test_api_polls_reco_random.py

```diff
@@ -0,0 +1,141 @@
+        self.assertEqual(len(resp.data["results"]), 3)
+
+        other_poll = Poll.objects.create(name="other")
+        video_5 = VideoFactory.create(make_safe_for_poll=other_poll)
+
+        resp = self.client.get(f"{self.url_path}?random=2")
+        results = resp.data["results"]
+        uids = [res["entity"]["uid"] for res in results]
+
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(results), 3)
+        self.assertNotIn(video_5.uid, uids)
+
+    def test_anon_can_list_with_limit(self):
+        """
+        An anonymous user can limit the size of the results by using the
+        `limit` query parameter.
+        """
+        resp = self.client.get(f"{self.url_path}?limit=1")
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(resp.data["results"]), 1)
+
+    def test_anon_can_list_videos_filtered_by_metadata(self):
+        """
+        Anonymous users can filter the recommended videos using a single
+        value filter.
+        """
+        resp = self.client.get(f"{self.url_path}?metadata[uploader]=_test_uploader_3")
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(resp.data["results"]), 1)
+        self.assertEqual(resp.data["results"][0]["entity"]["uid"], self.video_4.uid)
+
+        # filtering by an unknown upload must return an empty list
+        resp = self.client.get(f"{self.url_path}?metadata[uploader]=unknown_uploader")
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        self.assertEqual(resp.data["count"], 0)
+        self.assertEqual(resp.data["results"], [])
+
+    def test_anon_can_list_videos_filtered_by_pub_date(self):
+        """
+        Anonymous users can filter the recommended videos using a single
+        value filter.
+        """
+        resp = self.client.get(f"{self.url_path}?date_lte=2021-01-02")
+        results = resp.data["results"]
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(results), 1)
+        self.assertEqual(results[0]["entity"]["uid"], self.video_2.uid)
+
+        resp = self.client.get(f"{self.url_path}?date_lte=2021-01-03")
+        results = resp.data["results"]
+        uids = [res["entity"]["uid"] for res in results]
+
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(results), 2)
+        self.assertIn(self.video_2.uid, uids)
+        self.assertIn(self.video_3.uid, uids)
+
+        resp = self.client.get(f"{self.url_path}?date_gte=2021-01-01")
+        results = resp.data["results"]
+        uids = [res["entity"]["uid"] for res in results]
+
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(results), 3)
+        self.assertIn(self.video_2.uid, uids)
+        self.assertIn(self.video_3.uid, uids)
+        self.assertIn(self.video_4.uid, uids)
```

Comment:
Missing a test that actually checks that videos are ordered & selected randomly.

Can add 100 videos, then make two requests varying the ?random param and assert they return different lists. Assert they don't return the list 1,2,3,4,5 (that would be returned if ordered by scores).